src/components/data-exploration/differential-expression-tool/DiffExprCompute.jsx

```diff
@@ -91,21 +81,94 @@ const DiffExprCompute = (props) => {
       comparisonGroup[selectedComparison]['basis'] = `sample/${samples[0].key}`
     }
 
+    setSampleKeys(samples.map(sample => sample.key));
+
   }, [hierarchy, properties]);
 
-  const validateForm = () => {
+  const cellIdToSampleMap = useMemo(() => {
+    const mapping = [];
+    sampleKeys.forEach((key, idx) => {
+      const cellIds = properties[key].cellIds;
+      cellIds.forEach(cellId => mapping[cellId] = idx);
+    });
 
-    if (!comparisonGroup[selectedComparison]?.cellSet) {
-      setIsFormValid(false);
-      return;
+    return mapping;
+  }, [numSamples]);
+
+  // This function will return true if each of the compared groups is made up of at least
+  // 1 sample of with a enough cells (> MIN_NUM_CELLS_IN_GROUP).
```

Comment:
as soon as you change the name of `MIN_NUM_CELLS_IN_GROUP`, this comment becomes outdated. To avoid this, you can change the second line to `1 sample with more cells than a given minimum threshold`

Reply:
Ah you're right. I've corrected the comment.